### PR TITLE
Improving DaSiamRPN tracker sample

### PR DESCRIPTION
**Pull request by ieliz**
### Pull Request Readiness Checklist

I am trying to improve DaSiamRPN tracker sample:
1. Enhance code style and comments;
2. Re-writing tracker for compatibility with the [benchmark](https://github.com/opencv/opencv_contrib/pull/2516) (initialization of the tracker and update of the bounding box are changed).

- [x] I agree to contribute to the project under OpenCV (BSD) License.
- [x] To the best of my knowledge, the proposed patch is not based on a code under GPL or other license that is incompatible with OpenCV
- [x] The PR is proposed to proper branch
- [x] There is reference to original bug report and related work
- [ ] There is accuracy test, performance test and test data in opencv_extra repository, if applicable
      Patch to opencv_extra has the same branch name.
- [x] The feature is well documented and sample code can be built with the project CMake

### CODE REVIEW

File: samples/dnn/dasiamrpn_tracker.py [hunk 1]
```diff
@@ -1,11 +1,15 @@
...
+Original paper:
+https://arxiv.org/abs/1808.06048
+Link to original repo:
+https://github.com/foolwood/DaSiamRPN
+Link to the ONNX model of the network:
+https://www.dropbox.com/s/rr1lk9355vzolqv/dasiamrpn_model.onnx?dl=0
+Link to the ONNX model of the kernel_r1:
+https://www.dropbox.com/s/999cqx5zrfi7w4p/dasiamrpn_kernel_r1.onnx?dl=0
+Link to the ONNX model of the kernel_cls1:
+https://www.dropbox.com/s/qvmtszx5h339a0w/dasiamrpn_kernel_cls1.onnx?dl=0
```
Comment (dkurt) [inline samples/dnn/dasiamrpn_tracker.py:12]: Please avoid extra changes

File: samples/dnn/dasiamrpn_tracker.py [hunk 2]
```diff
@@ -81,13 +103,17 @@ def __generate_anchor(self):
         score_sz = int(self.score_size)
         self.anchor = np.tile(self.anchor, score_sz * score_sz).reshape((-1, 4))
         ori = - (score_sz / 2) * self.total_stride
-        xx, yy = np.meshgrid([ori + self.total_stride * dx for dx in range(score_sz)], [ori + self.total_stride * dy for dy in range(score_sz)])
-        xx, yy = np.tile(xx.flatten(), (self.anchor_num, 1)).flatten(), np.tile(yy.flatten(), (self.anchor_num, 1)).flatten()
-        self.anchor[:, 0], self.anchor[:, 1] = xx.astype(np.float32), yy.astype(np.float32)
```
Comment (dkurt) [inline samples/dnn/dasiamrpn_tracker.py:86]: Please do not change the lines without actual difference - it only makes review harder.

File: samples/dnn/dasiamrpn_tracker.py [hunk 3]
```diff
@@ -14,38 +18,62 @@
...
         self.avg_chans = np.mean(im, axis=(0, 1))
-        self.net = net
-        self.score = []
-
-        if ((self.target_sz[0] * self.target_sz[1]) / float(self.im_h * self.im_w)) < 0.004:
-             raise AssertionError("Initializing BB is too small-try to restart tracker with larger BB")
+        # Loading network`s and kernel`s models
+        net = cv.dnn.readNet(net)
+        kernel_r1 = cv.dnn.readNet(kernel_r1)
+        kernel_cls1 = cv.dnn.readNet(kernel_cls1)
```
Comment (dkurt) [inline samples/dnn/dasiamrpn_tracker.py:57]: Is that possible to reinitialize a tracker without model reloading?

File: samples/dnn/dasiamrpn_tracker.py (hunk 2 above)
Comment (ieliz) [reply samples/dnn/dasiamrpn_tracker.py:86]: I am sorry for the inconvenient. I thought it should be helpful to improve code style of the sample.
For now, should I reverse these changes (also, changes in the previous comment)?

File: samples/dnn/dasiamrpn_tracker.py (hunk 3 above)
Comment (ieliz) [reply samples/dnn/dasiamrpn_tracker.py]: I tried to load models once, and as I understand, it is worked with [the LaSOT benchmark](https://github.com/opencv/opencv_contrib/pull/2516).

File: samples/dnn/dasiamrpn_tracker.py (hunk 2 above)
Comment (dkurt) [reply samples/dnn/dasiamrpn_tracker.py:86]: Just add a new commit which returns it to origin state. Check `git diff 3.4..dasiamrpn` to see how it looks finally.

File: samples/dnn/dasiamrpn_tracker.py (hunk 3 above)
Comment (dkurt) [reply samples/dnn/dasiamrpn_tracker.py]: So, please move model loading into constructor (with model paths arguments).

File: samples/dnn/dasiamrpn_tracker.py [hunk 4]
```diff
@@ -195,23 +211,19 @@ def __get_subwindow_tracking(self, im, model_size, original_sz):
...
-
         return im_patch_original
 
-#function for reading paths, bounding box drawing, showing results
+# Sample for using DaSiamRPN tracker
 def main():
     parser = argparse.ArgumentParser(description="Run tracker")
-    parser.add_argument("--net", type=str, default="dasiamrpn_model.onnx", help="Full path to onnx model of net")
-    parser.add_argument("--kernel_r1", type=str, default="dasiamrpn_kernel_r1.onnx", help="Full path to onnx model of kernel_r1")
-    parser.add_argument("--kernel_cls1", type=str, default="dasiamrpn_kernel_cls1.onnx", help="Full path to onnx model of kernel_cls1")
```
Comment (dkurt) [inline samples/dnn/dasiamrpn_tracker.py:206]: Why these arguments have been removed? We need to let user control the paths.

File: samples/dnn/dasiamrpn_tracker.py [hunk 5]
```diff
@@ -257,17 +264,17 @@ def get_bb(event, x, y, flag, param):
...
-    #tracking loop
+    # Tracking loop
     while cap.isOpened():
         has_frame, frame = cap.read()
         if not has_frame:
             sys.exit(0)
-        tracker.track(frame)
-        w, h = tracker.target_sz
-        cx, cy = tracker.target_pos
+        update_status, new_bb = tracker.update(frame)
```
Comment (dkurt) [inline samples/dnn/dasiamrpn_tracker.py:276]:
```pyhton
_, new_bb = tracker.update(frame)
```

File: samples/dnn/dasiamrpn_tracker.py (hunk 4 above)
Comment (ieliz) [reply samples/dnn/dasiamrpn_tracker.py:206]: I moved it into parameters of the constructor - we can write paths during creating tracker.

File: samples/dnn/dasiamrpn_tracker.py (hunk 5 above)
Comment (ieliz) [reply samples/dnn/dasiamrpn_tracker.py:276]: OK

File: samples/dnn/dasiamrpn_tracker.py (hunk 4 above)
Comment (dkurt) [reply samples/dnn/dasiamrpn_tracker.py:206]: 1. That should be done for users who wants to run a script but cannot modify it or don't know how to do it
2. We have these flags in initial version so it'll be a regression

Comment (ieliz) [reply samples/dnn/dasiamrpn_tracker.py:206]: I understand that you are talking about the sample. But I want to change class for further more convenient use as an independent class. As an example, I used GOTURN from opencv-contrib- all models paths inside the tracker, not in the sample.
Maybe it is a regression for one specific sample, but IMHO, it is better for the tracker.

Comment (dkurt) [reply samples/dnn/dasiamrpn_tracker.py:206]: I don't see any problem to have both class level API and sample level API:
```python
class DaSiamRPNTracker:
    # Initialization of used values, initial bounding box, used network
    def __init__(self, net="dasiamrpn_model.onnx", kernel_r1="dasiamrpn_kernel_r1.onnx", kernel_cls1="dasiamrpn_kernel_cls1.onnx"):
```
```python
def main():
    parser = argparse.ArgumentParser(description="Run tracker")
    parser.add_argument("--net", type=str, default="dasiamrpn_model.onnx", help="Full path to onnx model of net")
    parser.add_argument("--kernel_r1", type=str, default="dasiamrpn_kernel_r1.onnx", help="Full path to onnx model of kernel_r1")
    parser.add_argument("--kernel_cls1", type=str, default="dasiamrpn_kernel_cls1.onnx", help="Full path to onnx model of kernel_cls1")

tracker = DaSiamRPNTracker(args.net, args.kernel_r1, args.cls1)
```

Comment (ieliz) [reply samples/dnn/dasiamrpn_tracker.py:206]: OK.
Sorry, I misunderstood your proposals.

**Pull request merged**